crates/rpc/rpc/src/eth/error.rs

```diff
@@ -328,6 +332,15 @@ impl From<PoolError> for EthApiError {
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+pub(crate) enum SignError {
+    #[error("Could not sign")]
+    CouldNotSign,
+    #[error("Unknown account")]
+    NoAccount,
+    #[error("Given typed data is not valid")]
+    TypedData,
```

Comment:
are these the standard RPC Error messages that e.g. Geth returns?

Reply:
The Unknown account yes, don't know about the other variants.

please add some docs here and check geth compact @fkrause98

Reply:
I am p sure Matt means "compat", i.e. check Geth's errors and ensure there's 1:1 compatibility with the messages

Reply:
From what I've seen, the only one we have in common is `NoAccount`. Both `CouldNotSign` and `TypedData` are equivalent to a lot of detailed errors in geth (with their messages specifying the exact cause for the error) and would need to be translated from the ones we have in the underlying implementation.

I'll add some docs. Should I make `CouldNotSign` and `TypedData` wrap the more specific errors `secp256k1::Error` and `Eip712Error`?